pyogrio/raw.py

```diff
@@ -1,6 +1,8 @@
 import warnings
 import os
 
+from osgeo_utils.auxiliary.util import GetOutputDriversFor
```

Comment:
I'd rather avoid this dependency and only rely on the parts of GDAL that we are binding against in Cython.

Looking at the [implementation](https://github.com/OSGeo/gdal/blob/master/swig/python/gdal-utils/osgeo_utils/auxiliary/util.py#L62) in GDAL, it looks like it is mostly a matter of iterating over drivers that support write and checking the driver's `GDAL_DMD_EXTENSION` [metadata item](https://github.com/OSGeo/gdal/blob/master/swig/python/gdal-utils/osgeo_utils/auxiliary/util.py#L57).

From there, it seems like we could handle this in a couple different ways:
- build a map of extensions to drivers when pyogrio is first initialized (roughly similar to how we use `DRIVERS` now)
- do the driver lookup at runtime based on the extension of a specific path

Reply:
In fact, it might be possible to make some minor modifications to `_ogr.pyx::ogr_list_drivers()` to return both the read / write capability of a driver as well as associated extensions, which we could then use to create an inverse map of extension to driver.

Reply:
Agreed that we don't want to depend on the GDAL python bindings (that would be an additional dependency, which also has no wheels, and thus would defeat the purpose of our wheels as a way to get GDAL).

As Brendan mentioned, I think we can indeed use the metadata from GDAL, now that we have bindings to getting that information (`_get_driver_metadata_item`). We have an open issue for this feature that mentions this as well (https://github.com/geopandas/pyogrio/issues/220), and that issue also links to the equivalent PR in fiona to add this feature (https://github.com/Toblerity/Fiona/pull/948), that might be closer to our code to look at compared to the GDAL implementation.

Reply:
Yes, I also already had a look at the implementation in gdal and it is indeed mainly based on the drivers `DMD_EXTENSIONS` metadata, with some extra logic.

I'm wondering, I also encountered a similar C++ function with even some more logic in it, would it be an option that is gets exposed in the C API. This could deduplicate the logic in gdal itself as wel as make it reusable in fiona, rasterio, pyogrio,... [communutils.cpp](https://github.com/OSGeo/gdal/blob/949ac934a6518f0e4fe60ca55827ec5e9d29d755/apps/commonutils.cpp#L69)?

Reply:
When reviewing #270 I also noticed there is a bug in the python version of `GetOutputDriversFor` so I opened a ticket in GDAL to report this: https://github.com/OSGeo/gdal/issues/8277